python/cudf/cudf/_lib/csv.pyx

```diff
@@ -0,0 +1,338 @@
+        for tv in true_values:
+            read_csv_args_c.true_values.push_back(tv.encode())
+
+    if false_values is not None:
+        read_csv_args_c.false_values.reserve(len(false_values))
+        for fv in false_values:
+            read_csv_args_c.false_values.push_back(fv.encode())
+
+    if na_values is not None:
+        read_csv_args_c.na_values.reserve(len(na_values))
+        for nv in na_values:
+            read_csv_args_c.na_values.push_back(nv.encode())
+
+    read_csv_args_c.keep_default_na = keep_default_na
+    read_csv_args_c.na_filter = na_filter
+    read_csv_args_c.dayfirst=dayfirst
+
+    return read_csv_args_c
+
+
+def read_csv(
+    filepath_or_buffer,
+    lineterminator="\n",
+    quotechar='"',
+    quoting=0,
+    doublequote=True,
+    header="infer",
+    mangle_dupe_cols=True,
+    usecols=None,
+    sep=",",
+    delimiter=None,
+    delim_whitespace=False,
+    skipinitialspace=False,
+    names=None,
+    dtype=None,
+    skipfooter=0,
+    skiprows=0,
+    dayfirst=False,
+    compression="infer",
+    thousands=None,
+    decimal=".",
+    true_values=None,
+    false_values=None,
+    nrows=None,
+    byte_range=None,
+    skip_blank_lines=True,
+    parse_dates=None,
+    comment=None,
+    na_values=None,
+    keep_default_na=True,
+    na_filter=True,
+    prefix=None,
+    index_col=None,
+    **kwargs,
+):
+    """
+    Cython function to call into libcudf API, see `read_csv`.
+
+    See Also
+    --------
+    cudf.io.csv.read_csv
+    """
+
+    if not isinstance(filepath_or_buffer, (BytesIO, StringIO, bytes)):
+        if not os.path.isfile(filepath_or_buffer):
+            raise FileNotFoundError(
+                errno.ENOENT, os.strerror(errno.ENOENT), filepath_or_buffer
+            )
+
+    if isinstance(filepath_or_buffer, StringIO):
+        filepath_or_buffer = filepath_or_buffer.read().encode()
+    elif isinstance(filepath_or_buffer, str) and not os.path.isfile(
+        filepath_or_buffer
+    ):
+        filepath_or_buffer = filepath_or_buffer.encode()
+
+    cdef read_csv_args read_csv_arg_c = make_read_csv_args(
+        filepath_or_buffer, lineterminator, quotechar, quoting, doublequote,
+        header, mangle_dupe_cols, usecols, sep, delimiter, delim_whitespace,
+        skipinitialspace, names, dtype, skipfooter, skiprows, dayfirst,
+        compression, thousands, decimal, true_values, false_values, nrows,
+        byte_range, skip_blank_lines, parse_dates, comment, na_values,
+        keep_default_na, na_filter, prefix, index_col)
+
+    cdef table_with_metadata c_result
+    with nogil:
+        c_result = move(cpp_read_csv(read_csv_arg_c))
+
+    meta_names = [name.decode() for name in c_result.metadata.column_names]
+    df = cudf.DataFrame._from_table(Table.from_unique_ptr(
+        move(c_result.tbl),
+        column_names=meta_names
+    ))
+
+    if names is not None and isinstance(names[0], (int)):
+        df.columns = [int(x) for x in df.columns]
```

Comment:
Can we use `df._data` instead of `df.columns` here? `df.columns` is expensive to materialize the Pandas Index object.